src/test/java/com/kuzmins/service/TicketServiceTest.java

```diff
@@ -0,0 +1,273 @@
+    ticket.setUser(user);
+
+    Mockito.when(ticketRepository.findTicketByIdAndUserId(id, userId)).thenReturn(ticket);
+
+    Ticket byId = ticketService.findByIdAndUserId(id, userId);
+
+    Assertions.assertEquals(ticket, byId);
+
+    Mockito.verify(ticketRepository).findTicketByIdAndUserId(id, userId);
+    Mockito.verifyNoMoreInteractions(ticketRepository);
+  }
+
+  @Test
+  void doNotFindTicketWhenCallFindByIdAndUserId() {
+    UUID id = UUID.randomUUID();
+    UUID userId = UUID.randomUUID();
+    Ticket ticket = new Ticket();
+
+    Mockito.when(ticketRepository.findTicketByIdAndUserId(id, userId)).thenReturn(null);
+    Ticket byId = ticketService.findByIdAndUserId(id, userId);
+
+    Assertions.assertNull(byId);
+
+    Mockito.verify(ticketRepository).findTicketByIdAndUserId(id, userId);
+    Mockito.verifyNoMoreInteractions(ticketRepository);
+  }
+
+  @Test
+  void throwExceptionWhenCallFindByIdAndUserId() {
+    UUID id = null;
+    UUID userId = null;
+
+    Mockito.when(ticketRepository.findTicketByIdAndUserId(id, userId))
+        .thenThrow(new IllegalArgumentException("id is null"));
+
+    IllegalArgumentException illegalArgumentException =
+        Assertions.assertThrows(
+            IllegalArgumentException.class, () -> ticketService.findByIdAndUserId(id, userId));
+
+    Assertions.assertNotNull(illegalArgumentException);
+    Assertions.assertEquals("id is null", illegalArgumentException.getMessage());
+
+    Mockito.verify(ticketRepository).findTicketByIdAndUserId(id, userId);
+    Mockito.verifyNoMoreInteractions(ticketRepository);
+  }
+
+  // ticketRepository.updateTicketType()
+  @Test
+  void updateTicketWhenCallUpdateTicketType() {
+    UUID id = UUID.randomUUID();
+    TicketType ticketType = TicketType.YEAR;
+    Ticket ticket = new Ticket(new User(), TicketType.MONTH);
+
+    Mockito.when(ticketRepository.findById(id)).thenReturn(Optional.of(ticket));
+
+    ticketService.updateTicketType(id, ticketType);
+
+    ArgumentCaptor<Ticket> argumentCaptor = ArgumentCaptor.forClass(Ticket.class);
+    Mockito.verify(ticketRepository).save(argumentCaptor.capture());
+    Ticket ticket1 = argumentCaptor.getValue();
+
+    Assertions.assertEquals(ticket.getId(), ticket1.getId());
+    Assertions.assertEquals(ticket.getUser(), ticket1.getUser());
+    Assertions.assertEquals(ticket.getTicketCreationTime(), ticket1.getTicketCreationTime());
+    Assertions.assertEquals(ticketType, ticket1.getType());
+
+    Mockito.verify(ticketRepository).findById(id);
+    Mockito.verifyNoMoreInteractions(ticketRepository);
+  }
+
+  @Test
+  void doNotUpdateWhenCallUpdateTicketType() {
+    UUID id = UUID.randomUUID();
+    TicketType ticketType = TicketType.YEAR;
+
+    Mockito.when(ticketRepository.findById(id)).thenReturn(Optional.empty());
+
+    ticketService.updateTicketType(id, ticketType);
+
+    Mockito.verify(ticketRepository).findById(id);
+    Mockito.verify(ticketRepository, Mockito.never()).save(any());
```

Comment:
avoid to use `any()`. Always try to use specific value for stubbed mock object

Reply:
I replaced any() to any(Ticket.class), because .findById() won't return any ticket. So I expect that nothing will be saved at all.